application/dash_application/utility/plot_builder.py

```diff
@@ -264,8 +267,8 @@ def is_empty_plot(trace_list) -> bool:
 
 
 # writing a factory may be peak Java poisoning but it might help with all these parameters
-def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, shapeby,
-             hovertext_cols, cutoff_lines: List[Tuple[str, float]]=[],
+def generate(title_text, sorted_data, y_fn, axis_text, colourby, shapeby,
+             hovertext_cols, mode, x_fn=None, cutoff_lines: List[Tuple[str, float]]=[],
```

Comment:
Though Python allows it, I would rather not keep the signature the same. My reasoning is that aside from accepting two different data types on the same formal parameter being unorthodox, it does less to communicate to other developers that the results of setting the Mode are the default behaviours, and defining the x_fn is an override rather than an equally valid option. Were they on equal footing I would fear confusion over why setting x_fn to 'lambda d: d[PINERY_COL.SampleName]' isn't equivalent to setting it to Mode.IUS. 
I have in this PR already adjusted every call to generate() to use this signature, no further refactoring is required.